network/endpoint_test.go

```diff
@@ -262,6 +262,37 @@ var _ = Describe("Test Endpoint", func() {
 				Expect(ep.Id).To(Equal(epInfo.Id))
 			})
 		})
+		Context("When secondary endpoint add failed", func() {
+			It("Should not be added to the network", func() {
+				_, ipnet, _ := net.ParseCIDR("0.0.0.0/0")
+				nw := &network{
+					Endpoints: map[string]*endpoint{},
+					Mode:      opModeTransparent,
+					extIf:     &externalInterface{BridgeName: "testbridge"},
+				}
+				epInfo := &EndpointInfo{
+					Id:      "768e8deb-eth1",
+					IfName:  eth0IfName,
+					NICType: cns.InfraNIC,
+				}
+				secondaryEpInfo := &EndpointInfo{
+					MacAddress: netio.BadHwAddr, // mock netlink will fail to set link state on bad eth
+					NICType:    cns.DelegatedVMNIC,
+					Routes:     []RouteInfo{{Dst: *ipnet}},
+				}
+				ep, err := nw.newEndpointImpl(nil, netlink.NewMockNetlink(false, ""), platform.NewMockExecClient(false),
+					netio.NewMockNetIO(false, 0), nil, NewMockNamespaceClient(), []*EndpointInfo{epInfo, secondaryEpInfo})
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(Equal("SecondaryEndpointClient Error: Mock Netlink Error"))
```

Comment:
nit: we can reuse the variable for error `netlink.ErrorMockNetlink`.

Reply:
sure, updated